sklearn/tests/test_pipeline.py

```diff
@@ -1828,5 +1828,35 @@ def test_routing_passed_metadata_not_supported(method):
         getattr(pipe, method)([[1]], sample_weight=[1], prop="a")
 
 
+@pytest.mark.usefixtures("enable_slep006")
+def test_pipeline_with_estimator_with_len():
+    """Test that pipeline works with estimators that have a `__len__` method."""
+
+    class LenTransformer(TransformerMixin, BaseEstimator):
+        def __init__(self, value):
+            self.value = value
+
+        def fit(self, X, y=None):
+            return self
+
+        def transform(self, X):
+            return X
+
+        def __len__(self):
+            return self.value
```

Comment:
I would propose to use `RandomTreesEmbedding` since it already implements `__len__` since this is an ensemble of trees.